### ISSUE TITLE
[Telecinco] Failed to download m3u8 information: HTTP Error 403: Forbidden

### ISSUE THREAD

**Issue by Juancafr** [yt-dlp/yt-dlp#10986]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

I want to download a program from the mitele website and it gives me this error.

### Provide verbose output that clearly demonstrates the problem

- [x] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [x] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
PS C:\yt-dllp> .\yt-dlp.exe -vU https://www.telecinco.es/el-rival-mas-debil/a-la-carta/20240911/programa-2-completo-video-11-septiembre_18_013427350.html -f best
[debug] Command-line config: ['-vU', 'https://www.telecinco.es/el-rival-mas-debil/a-la-carta/20240911/programa-2-completo-video-11-septiembre_18_013427350.html', '-f', 'best']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2024.08.06 from yt-dlp/yt-dlp [4d9231208] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.22631-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-116990-g4646a74d1e-20240911 (setts), ffprobe N-116990-g4646a74d1e-20240911
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.32.3, sqlite3-3.35.5, urllib3-2.2.2, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
WARNING: "-f best" selects the best pre-merged format which is often not the best option.
         To let yt-dlp download and merge the best available formats, simply do not pass any format selection.
         If you know what you are doing and want only the best pre-merged format, use "-f b" instead to suppress this warning
[debug] Loaded 1830 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2024.08.06 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2024.08.06 from yt-dlp/yt-dlp)
[Telecinco] Extracting URL: https://www.telecinco.es/el-rival-mas-debil/a-la-carta/20240911/programa-2-completo-video-11-septiembre_18_013427350.html
[Telecinco] programa-2-completo-video-11-septiembre_18_013427350: Downloading webpage
[Telecinco] 5de0091f-23ca-44f0-9a5c-1f18b607732f: Downloading config JSON
[Telecinco] 5de0091f-23ca-44f0-9a5c-1f18b607732f: Downloading JSON metadata
[Telecinco] 5de0091f-23ca-44f0-9a5c-1f18b607732f: Downloading JSON metadata
[Telecinco] 5de0091f-23ca-44f0-9a5c-1f18b607732f: Downloading JSON metadata
[Telecinco] 5de0091f-23ca-44f0-9a5c-1f18b607732f: Downloading m3u8 information
ERROR: [Telecinco] programa-2-completo-video-11-septiembre_18_013427350: Failed to download m3u8 information: HTTP Error 403: Forbidden (caused by <HTTPError 403: Forbidden>)
  File "yt_dlp\extractor\common.py", line 740, in extract
  File "yt_dlp\extractor\telecinco.py", line 138, in _real_extract
  File "yt_dlp\extractor\telecinco.py", line 102, in _parse_content
  File "yt_dlp\extractor\common.py", line 2019, in _extract_m3u8_formats
  File "yt_dlp\extractor\common.py", line 2041, in _extract_m3u8_formats_and_subtitles
  File "yt_dlp\extractor\common.py", line 960, in _download_webpage_handle
  File "yt_dlp\extractor\common.py", line 909, in _request_webpage

  File "yt_dlp\extractor\common.py", line 896, in _request_webpage
  File "yt_dlp\YoutubeDL.py", line 4165, in urlopen
  File "yt_dlp\networking\common.py", line 117, in send
  File "yt_dlp\networking\_helper.py", line 208, in wrapper
  File "yt_dlp\networking\common.py", line 340, in send
  File "yt_dlp\networking\_requests.py", line 365, in _send
yt_dlp.networking.exceptions.HTTPError: HTTP Error 403: Forbidden
```

**Comment by n0ttr** [comment]
The issue is that requests to `vbdk.mediaset.es` are returning `403` if http headers doesn't look like http headers sent by a web browser I can imagine.

I am not familiar with this codebase so I don't know which would be the better way of applying this fix.

I haven't taken the time to determine which headers are actually required, but this works at the moment:


file: `networking/common.py#336`
```python
    @wrap_request_errors
    def send(self, request: Request) -> Response:
        if not isinstance(request, Request):
            raise TypeError('Expected an instance of Request')
        
        if 'vbdk.mediaset.es' in request.url:
            request.headers = {
                'accept': '*/*',
                'accept-language': 'en-US,en;q=0.9,es-ES;q=0.8,es;q=0.7,ca;q=0.6',
                'cache-control': 'no-cache',
                'origin': 'https://www.mitele.es',
                'pragma': 'no-cache',
                'priority': 'u=1, i',
                'referer': 'https://www.mitele.es/programas-tv/horizonte/temporada-5/programa-169-40_013384375/player/',
                'sec-ch-ua': '"Chromium";v="128", "Not;A=Brand";v="24", "Google Chrome";v="128"',
                'sec-ch-ua-mobile': '?0',
                'sec-ch-ua-platform': '"Linux"',
                'sec-fetch-dest': 'empty',
                'sec-fetch-mode': 'cors',
                'sec-fetch-site': 'cross-site',
                'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36'
            }
        return self._send(request)
```
I was trying to download `https://www.mitele.es/programas-tv/horizonte/temporada-5/programa-169-40_013384375/player/` so if it is required the `referer` header has to be set accordingly to the url you want to download.

**Comment by Juancafr** [comment]
> El problema es que las solicitudes que `vbdk.mediaset.es`se devuelven `403`si los encabezados http no se parecen a los encabezados http enviados por un navegador web, me imagino.
> 
> No estoy familiarizado con esta base de código, por lo que no sé cuál sería la mejor manera de aplicar esta solución.
> 
> No me he tomado el tiempo de determinar qué encabezados son realmente necesarios, pero esto funciona en este momento:
> 
> archivo:`networking/common.py#336`
> 
> ```python
>     @wrap_request_errors
>     def send(self, request: Request) -> Response:
>         if not isinstance(request, Request):
>             raise TypeError('Expected an instance of Request')
>         
>         if 'vbdk.mediaset.es' in request.url:
>             request.headers = {
>                 'accept': '*/*',
>                 'accept-language': 'en-US,en;q=0.9,es-ES;q=0.8,es;q=0.7,ca;q=0.6',
>                 'cache-control': 'no-cache',
>                 'origin': 'https://www.mitele.es',
>                 'pragma': 'no-cache',
>                 'priority': 'u=1, i',
>                 'referer': 'https://www.mitele.es/programas-tv/horizonte/temporada-5/programa-169-40_013384375/player/',
>                 'sec-ch-ua': '"Chromium";v="128", "Not;A=Brand";v="24", "Google Chrome";v="128"',
>                 'sec-ch-ua-mobile': '?0',
>                 'sec-ch-ua-platform': '"Linux"',
>                 'sec-fetch-dest': 'empty',
>                 'sec-fetch-mode': 'cors',
>                 'sec-fetch-site': 'cross-site',
>                 'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36'
>             }
>         return self._send(request)
> ```
> 
> Estaba intentando descargarlo `https://www.mitele.es/programas-tv/horizonte/temporada-5/programa-169-40_013384375/player/`, así que si es necesario, el `referer`encabezado debe configurarse de acuerdo con la URL que desea descargar.



> The issue is that requests to `vbdk.mediaset.es` are returning `403` if http headers doesn't look like http headers sent by a web browser I can imagine.
> 
> I am not familiar with this codebase so I don't know which would be the better way of applying this fix.
> 
> I haven't taken the time to determine which headers are actually required, but this works at the moment:
> 
> file: `networking/common.py#336`
> 
> ```python
>     @wrap_request_errors
>     def send(self, request: Request) -> Response:
>         if not isinstance(request, Request):
>             raise TypeError('Expected an instance of Request')
>         
>         if 'vbdk.mediaset.es' in request.url:
>             request.headers = {
>                 'accept': '*/*',
>                 'accept-language': 'en-US,en;q=0.9,es-ES;q=0.8,es;q=0.7,ca;q=0.6',
>                 'cache-control': 'no-cache',
>                 'origin': 'https://www.mitele.es',
>                 'pragma': 'no-cache',
>                 'priority': 'u=1, i',
>                 'referer': 'https://www.mitele.es/programas-tv/horizonte/temporada-5/programa-169-40_013384375/player/',
>                 'sec-ch-ua': '"Chromium";v="128", "Not;A=Brand";v="24", "Google Chrome";v="128"',
>                 'sec-ch-ua-mobile': '?0',
>                 'sec-ch-ua-platform': '"Linux"',
>                 'sec-fetch-dest': 'empty',
>                 'sec-fetch-mode': 'cors',
>                 'sec-fetch-site': 'cross-site',
>                 'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36'
>             }
>         return self._send(request)
> ```
> 
> I was trying to download `https://www.mitele.es/programas-tv/horizonte/temporada-5/programa-169-40_013384375/player/` so if it is required the `referer` header has to be set accordingly to the url you want to download.

I'm a bit new to these topics, so could you explain what I should do? I don't know if I should modify any file, add a new one, or what..thanks

**Comment by n0ttr** [comment]
@Juancafr 

He dejado el mensaje para los que mantienen el repositorio que sepan como solucionar.
Si quieres aplicar tu el fix manualmente tienes que encontrar donde tienes instalado `yt-dlp`
Por ejemplo, en linux a mi se me ha instalado en esta carpeta `/usr/lib/python3.12/site-packages/yt_dlp/utils` pero variara dependiendo de tu systema operativo.

Una vez hayas encontrado la carpeta busca el archivo `networking/common.py#336`
es decir, dentro de la carpeta `networking` el archivo `common.py` y vas a la linea `336` (esto es teniendo en cuenta que tienes la ultima version de `yt-dlp` en tu sistema)
ahi encontraras la siguiente funcion, si no tenemos la misma version igual no esta en la linea `336`
```python
@wrap_request_errors
    def send(self, request: Request) -> Response:
        if not isinstance(request, Request):
            raise TypeError('Expected an instance of Request')
      
        return self._send(request)
```
pues bien, reemplaza esa funcion por este codigo
```python
@wrap_request_errors
    def send(self, request: Request) -> Response:
        if not isinstance(request, Request):
            raise TypeError('Expected an instance of Request')
        
        if 'vbdk.mediaset.es' in request.url:
            request.headers = {
                'accept': '*/*',
                'accept-language': 'en-US,en;q=0.9,es-ES;q=0.8,es;q=0.7,ca;q=0.6',
                'cache-control': 'no-cache',
                'origin': 'https://www.mitele.es/',
                'pragma': 'no-cache',
                'priority': 'u=1, i',
                'referer': request.url,
                'sec-ch-ua': '"Chromium";v="128", "Not;A=Brand";v="24", "Google Chrome";v="128"',
                'sec-ch-ua-mobile': '?0',
                'sec-ch-ua-platform': '"Linux"',
                'sec-fetch-dest': 'empty',
                'sec-fetch-mode': 'cors',
                'sec-fetch-site': 'cross-site',
                'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36'
            }
        return self._send(request)
```

Eso son los pasos que me han funcionado a mi, es un parche guarrero y deberia solucionarse de otra manera mejor, pero asi funciona de momento

**Comment by grqz** [comment]
@n0ttr i'm geo-blocked, could you please test this patch for me?
```diff
diff --git a/yt_dlp/extractor/telecinco.py b/yt_dlp/extractor/telecinco.py
index 7a9dcd71c..c00f2016c 100644
--- a/yt_dlp/extractor/telecinco.py
+++ b/yt_dlp/extractor/telecinco.py
@@ -1,12 +1,12 @@
 import json
-import re
 
 from .common import InfoExtractor
 from ..utils import (
     clean_html,
     int_or_none,
     str_or_none,
-    try_get,
+    traverse_obj,
+    url_or_none,
 )
 
 
@@ -91,8 +91,8 @@ def _parse_content(self, content, url):
         stream = caronte['dls'][0]['stream']
         headers = self.geo_verification_headers()
         headers.update({
-            'Content-Type': 'application/json;charset=UTF-8',
-            'Origin': re.match(r'https?://[^/]+', url).group(0),
+            'Origin': url_or_none(url),
+            'Referer': url_or_none(url),
         })
         cdn = self._download_json(
             caronte['cerbero'], video_id, data=json.dumps({
@@ -100,7 +100,7 @@ def _parse_content(self, content, url):
                 'gbx': self._download_json(services['gbx'], video_id)['gbx'],
             }).encode(), headers=headers)['tokens']['1']['cdn']
         formats = self._extract_m3u8_formats(
-            stream + '?' + cdn, video_id, 'mp4', 'm3u8_native', m3u8_id='hls')
+            stream + '?' + cdn, video_id, 'mp4', 'm3u8_native', m3u8_id='hls', headers=headers)
 
         return {
             'id': video_id,
@@ -121,7 +121,7 @@ def _real_extract(self, url):
         if article.get('editorialType') != 'VID':
             entries = []
             body = [article.get('opening')]
-            body.extend(try_get(article, lambda x: x['body'], list) or [])
+            body.extend(traverse_obj(article, ('body', {list})))
             for p in body:
                 if not isinstance(p, dict):
                     continue
```

**Comment by Juancafr** [comment]
In my case it is Windows 11, therefore it is not an installation but an exe that I use with a command line. I don't know how I should act here

**Comment by Juancafr** [comment]
> > In my case it is Windows 11, therefore it is not an installation but an exe that I use with a command line. I don't know how I should act here
> 
> you can choose from the following:
> 
> 1. wait for [[ie/Telecinco] fix extractor #10999](https://github.com/yt-dlp/yt-dlp/pull/10999) to be merged
> 2. apply patch and run from source code
> 3. apply patch and build yt-dlp binaries

I'll wait because I wouldn't even know where to start. Thanks!

**Comment by Juancafr** [comment]
Any news?

**Comment by Juancafr** [comment]
The problem is that I am on Windows and I don't have that file. In my case I run an .exe file in a command window, nothing more
________________________________
De: n0ttr ***@***.***>
Enviado: viernes, 13 de septiembre de 2024 8:46
Para: yt-dlp/yt-dlp ***@***.***>
Cc: Juancafr ***@***.***>; Author ***@***.***>
Asunto: Re: [yt-dlp/yt-dlp] [Telecinco] Failed to download m3u8 information: HTTP Error 403: Forbidden (Issue #10986)


The issue is that requests to vbdk.mediaset.es are returning 403 if http headers doesn't look like http headers sent by a web browser I can imagine.

I am not familiar with this codebase so I don't know which would be the better way of applying this fix.

I haven't taken the time to determine which headers are actually required, but this works at the moment:

file: networking/common.py#336

    @wrap_request_errors
    def send(self, request: Request) -> Response:
        if not isinstance(request, Request):
            raise TypeError('Expected an instance of Request')

        if 'vbdk.mediaset.es' in request.url:
            request.headers = {
                'accept': '*/*',
                'accept-language': 'en-US,en;q=0.9,es-ES;q=0.8,es;q=0.7,ca;q=0.6',
                'cache-control': 'no-cache',
                'origin': 'https://www.mitele.es',
                'pragma': 'no-cache',
                'priority': 'u=1, i',
                'referer': 'https://www.mitele.es/programas-tv/horizonte/temporada-5/programa-169-40_013384375/player/',
                'sec-ch-ua': '"Chromium";v="128", "Not;A=Brand";v="24", "Google Chrome";v="128"',
                'sec-ch-ua-mobile': '?0',
                'sec-ch-ua-platform': '"Linux"',
                'sec-fetch-dest': 'empty',
                'sec-fetch-mode': 'cors',
                'sec-fetch-site': 'cross-site',
                'user-agent': 'Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36'
            }
        return self._send(request)

I was trying to download https://www.mitele.es/programas-tv/horizonte/temporada-5/programa-169-40_013384375/player/ so if it is required the referer header has to be set accordingly to the url you want to download.

**Comment by luisignaciomenendez** [comment]
> Any news?

You can apply the patch above manually, just the path is different for you. Try going to  (you might need to uncover hidden files if you're on mac):

/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/yt_dlp/networking/common.py

If you change the function mentioned by @n0ttr and save it worked for me!

**Comment by Juancafr** [comment]
> > Any news?
> 
> You can apply the patch above manually, just the path is different for you. Try going to (you might need to uncover hidden files if you're on mac):
> 
> /Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/yt_dlp/networking/common.py
> 
> If you change the function mentioned by @n0ttr and save it worked for me!

I'm on Windows

**Comment by ecarvalho123** [comment]
I'm having some trouble locating the /networking/ directory where common.py is located. Running MINT Linux and my install location follows:

`eduardo@ares:/usr/lib/python3/dist-packages/yt_dlp$ ll
total 716
drwxr-xr-x   6 root root   4096 Jan 16  2024 ./
drwxr-xr-x 254 root root  20480 Sep 11 23:16 ../
-rw-r--r--   1 root root  21157 Apr  8  2022 aes.py
-rw-r--r--   1 root root   3128 Apr  8  2022 cache.py
-rw-r--r--   1 root root   9275 Apr  8  2022 compat.py
-rw-r--r--   1 root root  37785 Apr  8  2022 cookies.py
drwxr-xr-x   3 root root   4096 Jan 16  2024 downloader/
drwxr-xr-x   4 root root  36864 Jan 16  2024 extractor/
-rw-r--r--   1 root root  38414 Apr  9  2022 __init__.py
-rw-r--r--   1 root root  21594 Apr  8  2022 jsinterp.py
-rw-r--r--   1 root root    452 Apr  8  2022 __main__.py
-rw-r--r--   1 root root   5241 Apr  8  2022 minicurses.py
-rw-r--r--   1 root root  85013 Apr  9  2022 options.py
drwxr-xr-x   3 root root   4096 Jan 16  2024 postprocessor/
drwxr-xr-x   2 root root   4096 Jun 27 19:59 __pycache__/
-rw-r--r--   1 root root   8861 Apr  8  2022 socks.py
-rw-r--r--   1 root root  10327 Apr  8  2022 update.py
-rw-r--r--   1 root root 183561 Apr  8  2022 utils.py
-rw-r--r--   1 root root    108 Apr  8  2022 version.py
-rw-r--r--   1 root root  11161 Apr  8  2022 webvtt.py
-rw-r--r--   1 root root 185355 Apr  8  2022 YoutubeDL.py
`

Thank you for any tips in advance.

**Comment by bashonly** [comment]
do NOT edit `yt_dlp/networking/common.py`, that is not the correct approach to solving this issue; doing so will break countless other extractors/sites

**Comment by Juancafr** [comment]
no news?

**Comment by emv33** [comment]
#10999 solves the problem  for telecinco.es, but not for mitele.es 

The following video will fail to download even with #10999 applied: https://www.mitele.es/programas-tv/de-viernes/temporada-1/programa-44-40_013481363/player/

**Comment by bashonly** [comment]
@emv33 open a new issue with complete verbose output for mitele.es please

**Comment by DarkZeros** [comment]
I made a more generic version to fix this issue here: https://github.com/yt-dlp/yt-dlp/pull/11142
It works for me to download from mitele trough a proxy properly.

**Comment by Juancafr** [comment]
> Hice una versión más genérica para solucionar este problema aquí: #11142 Me funciona para descargar desde mitele a través de un proxy correctamente.



> I made a more generic version to fix this issue here: #11142 It works for me to download from mitele trough a proxy properly.

Can you explain to me what I should go down? I just don't understand it, I'm a newbie to these topics. I used to use the .exe in Windows with a command line to download mitele, but it hasn't worked for me for weeks and I don't know how I should solve it.
